state/allwatcher.go

```diff
@@ -1573,9 +1573,8 @@ func (b *allModelWatcherStateBacking) Changed(all *multiwatcherStore, change wat
 
 	st, err := b.getState(modelUUID)
 	if err != nil {
-		if exists, modelErr := b.st.ModelExists(modelUUID); exists && modelErr == nil {
-			// The entity's model is gone so remove the entity
-			// from the store.
+		if errors.IsNotFound(err) {
```

Comment:
This is the important part to stop the all-watcher exiting.

A not-found error would previously bubble up stopping the watcher, which it shouldn't - the watcher now just deals with the fact that the model is gone.